macosx*_arm64: m4ri fails to build (png.h)

```
  [m4ri-20200125]   [spkg-install] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./m4ri -g -O2 -I/Users/runner/work/passagemath/passagemath/local/include -g -fPIC -Wall -pedantic -c m4ri/io.c  -fno-common -DPIC -o m4ri/.libs/io.o
  [m4ri-20200125]   [spkg-install] 2 warnings generated.
  [m4ri-20200125]   [spkg-install] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./m4ri -g -O2 -I/Users/runner/work/passagemath/passagemath/local/include -g -fPIC -Wall -pedantic -c m4ri/djb.c  -fno-common -DPIC -o m4ri/.libs/djb.o
  [m4ri-20200125]   [spkg-install] /bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I./m4ri       -g -O2 -I/Users/runner/work/passagemath/passagemath/local/include -g -fPIC -Wall -pedantic -c -o m4ri/mp.lo m4ri/mp.c
  [m4ri-20200125]   [spkg-install] m4ri/io.c:28:10: fatal error: 'png.h' file not found
  [m4ri-20200125]   [spkg-install] #include <png.h>
  [m4ri-20200125]   [spkg-install]          ^~~~~~~
```
https://github.com/passagemath/passagemath/actions/runs/11394855901/job/31706166148#step:8:3485

--- a/.github/workflows/dist.yml
+++ b/.github/workflows/dist.yml
@@ -209,9 +209,9 @@ jobs:
     env:
       CAN_DEPLOY: ${{ secrets.SAGEMATH_PYPI_API_TOKEN != '' }}
       # SPKGs to install as system packages
-      SPKGS: _bootstrap _prereq openssl curl readline libpng info ${{ startsWith(matrix.os, 'macos') && 'gfortran' || '' }}
+      SPKGS: _bootstrap _prereq openssl curl readline info ${{ startsWith(matrix.os, 'macos') && 'gfortran' || '' }}
       # Non-Python packages to install as spkgs
-      TARGETS_PRE: ccache gmp mpfr mpc bliss coxeter3 mcqd meataxe sirocco boost_cropped tdlib brial giac glpk libhomfly gsl pari ntl flint linbox iml lcalc libbraiding m4ri m4rie planarity rw symmetrica singular eclib gap ecl libgd
+      TARGETS_PRE: ccache libpng gmp mpfr mpc bliss coxeter3 mcqd meataxe sirocco boost_cropped tdlib brial giac glpk libhomfly gsl pari ntl flint linbox iml lcalc libbraiding m4ri m4rie planarity rw symmetrica singular eclib gap ecl libgd
       CIBW_BUILD: "*${{ matrix.build }}*"
       # Disable building PyPy wheels on all platforms
       CIBW_SKIP: "pp*"
@@ -282,7 +282,7 @@ jobs:
           ./configure --with-python="${{ steps.python.outputs.python-path }}" --prefix=$HOME/runner-venv
           make cibuildwheel
           export PATH=build/bin:$PATH
-          echo CIBW_BEFORE_ALL="( $(sage-print-system-package-command debian --yes --no-install-recommends install $(sage-get-system-packages debian $SPKGS)) || $(sage-print-system-package-command fedora --yes --no-install-recommends install $(sage-get-system-packages fedora $SPKGS | sed s/pkg-config/pkgconfig/)) || ( $(sage-print-system-package-command homebrew --yes --no-install-recommends install $(sage-get-system-packages homebrew $SPKGS)) || $(sage-print-system-package-command alpine --yes --no-install-recommends install $(sage-get-system-packages alpine $SPKGS)) || echo error ignored) ) && rm -f config.status && PREFIX=${{ startsWith(matrix.os, 'ubuntu') && '/host/sage-\$AUDITWHEEL_PLAT' || '\$HOME/sage-local' }} && if cp \$PREFIX/config.status . 2>/dev/null; then chmod +x config.status; fi && if [ -x ./config.status ]; then ./config.status; else ./configure --enable-build-as-root --prefix=\$PREFIX --with-system-gfortran=force && cp config.status prefix/; fi && (MAKE=\"make -j6\" make -k V=0 $TARGETS_PRE || echo error ignored) && (echo \"passagemath-conf @ file://\$(pwd)/pkgs/sage-conf\" && echo \"passagemath-setup @ file://\$(pwd)/pkgs/sage-setup\") > constraints.txt" >> "$GITHUB_ENV"
+          echo CIBW_BEFORE_ALL="( $(sage-print-system-package-command debian --yes --no-install-recommends install $(sage-get-system-packages debian $SPKGS)) || $(sage-print-system-package-command fedora --yes --no-install-recommends install $(sage-get-system-packages fedora $SPKGS | sed s/pkg-config/pkgconfig/)) || ( $(sage-print-system-package-command homebrew --yes --no-install-recommends install $(sage-get-system-packages homebrew $SPKGS)) || $(sage-print-system-package-command alpine --yes --no-install-recommends install $(sage-get-system-packages alpine $SPKGS)) || echo error ignored) ) && rm -f config.status && PREFIX=${{ startsWith(matrix.os, 'ubuntu') && '/host/sage-\$AUDITWHEEL_PLAT' || '\$HOME/sage-local' }} && if cp \$PREFIX/config.status . 2>/dev/null; then chmod +x config.status; fi && if [ -x ./config.status ]; then ./config.status; else ./configure --enable-build-as-root --prefix=\$PREFIX --with-system-gfortran=force --without-system-libpng && cp config.status prefix/; fi && (MAKE=\"make -j6\" make -k V=0 $TARGETS_PRE || echo error ignored) && (echo \"passagemath-conf @ file://\$(pwd)/pkgs/sage-conf\" && echo \"passagemath-setup @ file://\$(pwd)/pkgs/sage-setup\") > constraints.txt" >> "$GITHUB_ENV"
           mkdir -p unpacked
           for sdist in dist/$pkg*.tar.gz; do
               (cd unpacked && tar xfz - ) < $sdist

--- a/.github/workflows/dist.yml
+++ b/.github/workflows/dist.yml
@@ -279,6 +279,7 @@ jobs:
         # omit sagemath-{meataxe,sirocco} for now -- needs sagemath-modules
         run: |
           "${{ steps.python.outputs.python-path }}" -m pip install setuptools
+          . ./.homebrew-build-env
           ./configure --with-python="${{ steps.python.outputs.python-path }}" --prefix=$HOME/runner-venv
           make cibuildwheel
           export PATH=build/bin:$PATH

